### PR TITLE
Sub nav updates:

### PR DESCRIPTION
**Pull request by wen-2018**
1. Update /firefox to have consistent sub navs
2. Remove and re-order links on /firefox/more page

## Issue / Bugzilla link

Fixes https://github.com/mozilla/bedrock/issues/14718

### Local server

- [ ] http://localhost:8000/en-US/firefox/
- [ ] http://localhost:8000/en-US/firefox/all/
- [ ] http://localhost:8000/en-US/firefox/enterprise/
- [ ] http://localhost:8000/en-US/firefox/browsers/mobile/
- [ ] http://localhost:8000/en-US/firefox/browsers/mobile/android/
- [ ] http://localhost:8000/en-US/firefox/browsers/mobile/ios/
- [ ] http://localhost:8000/en-US/firefox/browsers/mobile/focus/
- [ ] http://localhost:8000/en-US/firefox/browsers/mobile/get-app/
- [ ] http://localhost:8000/en-US/firefox/facebookcontainer/
- [ ] http://localhost:8000/en-US/firefox/more/

### Demo server

- [ ]  https://www-demo7.allizom.org/en-US/firefox/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/all/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/enterprise/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/browsers/mobile/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/browsers/mobile/android/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/browsers/mobile/ios/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/browsers/mobile/focus/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/browsers/mobile/get-app/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/facebookcontainer/
- [ ]  https://www-demo7.allizom.org/en-US/firefox/more/

### CODE REVIEW

File: l10n/en/firefox/more.ftl
Comment (reemhamz) [inline l10n/en/firefox/more.ftl:45]: This string can be removed since it's not being used anywhere

File: bedrock/firefox/templates/firefox/more.html
Comment (reemhamz) [inline bedrock/firefox/templates/firefox/more.html:1]: **Question:** Should we also be adding the subnav to this page?

**Review by reemhamz — CHANGES_REQUESTED** [review]
Good job getting this done, you've done it pretty smoothly!
I just have a couple of questions regarding some files, and I think we should push this to a demo URL so Adria could have a look before we merge. I can take care of that while you're on PTO 👍🏼 I'll approve once some changes are done and Adria gives us the go-ahead

File: bedrock/firefox/templates/firefox/includes/sub-nav-firefox.html
Comment (janbrasna) [reply bedrock/firefox/templates/firefox/includes/sub-nav-firefox.html:1]: I think it differs slightly wrt the page title/icon:

Screenshot 2024-07-02 at 20 50 38

but yea, agree it should be just "Firefox", as the "Mobile" is then the currently selected menu item of the actual subnav on the right, so keeping all the subnavs from just one source file for consistency ❤️

File: bedrock/firefox/templates/firefox/more.html
Comment (janbrasna) [reply bedrock/firefox/templates/firefox/more.html:1]: The subnav is missing in these places: _(where it also often links to… and getting back from there is a pita…)_

- `firefox/features` (and all features subpages too)
- `firefox/browsers/compare` (there the actual comparisons then have their own submenu)
- `firefox/faq` has its own submenu that'd be worth updating with this too
- `firefox/browsers/what-is-a-browser` and `firefox/browsers/browser-history` then kinda reuse the faq submenu but again in a way the IA for browsing back and forth doesn't work, it all leads deeper or away, to isolated silos of pages that don't offer a good hierarchy how to navigate them, and eventually return someplace else. So including these in the subnav overhaul would be awesome 🚀 

Actually the "FAQ" page could itself be a good addition to the subnav, as it's not linked from anywhere visible.

Comment (janbrasna) [reply bedrock/firefox/templates/firefox/more.html:1]: OTOH there's one tiny issue now introduced with the default `firefox` subnav being also used for `firefox/enterprise` (which previously was on the beta/developer subnav) — the actual "firefox/enterprise" page is not part of this subnav here, therefore you can't get to it via navigation, and can't see the page bolded (aria current page role) when you're there, as it's not in the subnav at all.

I don't think the previous enterprise grouping with beta, dev, mobile beta & TestFlight pages in one subnav was too bad, esp. that the IA in footer also groups ESR with dev and other channels so at least that felt consistent. I never saw the brief so this might have been decided already, but if there's still some undefined leeway in possibility to better group the other channels together, this might be a good opportunity to fix up also their distinct submenu (which, like the issue being tamed here, has the items all over the place when navigating around), so that one could also undergo a little cleanup, making all the items for pages as:
1. https://www.mozilla.org/firefox/enterprise/
1. https://www.mozilla.org/firefox/developer/
1. https://www.mozilla.org/firefox/channel/desktop/
1. https://www.mozilla.org/firefox/channel/android/
1. https://www.mozilla.org/firefox/channel/ios/

one neat consistent submenu for channels.

Comment (reemhamz) [reply bedrock/firefox/templates/firefox/more.html:1]: Hey @janbrasna -- as usual, your findings are always super impactful to our decision-making‼️
I brought up the fact that the subnav should be on the `firefox/browsers/compare` and `firefox/features` pages and the team agreed and Wen has updated that part. 

However, would you mind explaining why the `firefox/faq` and its connected page in this subnav update? I think it would be worthwhile making new subnavs for these pages rather than use the one that was created in this code update.

Regarding the `firefox/enterprise` page, alongside its old subnav + its connected pages, I brought this up with the SEO manager and will see what they say 👍🏼 as I think it would be worth it to have a discussion/see the reasoning behind these pages.

Comment (janbrasna) [reply bedrock/firefox/templates/firefox/more.html:1]: Oh I should have separated out what's probably worth taking care of with this changeset and is actionable, from just thinking out loud about all the interconnected content silos with their own local subnavs that I wanted to actually take a look at improving someday, as I regularly get lost in the IA;)

I will explain it better in an issue once I do a little bit of experimenting to see what would help, translating it into some form of drawing mapping out the IA compared to the local navigation content provided. But just to quickly follow up:
- The Enterprise vs. channels is a tricky one as Enterprise is one of the product options for /firefox, but is not in the nav there; it was always grouped in the navigation (=see footer) together with channels. But it's good to have more ways of organising the content to cater to different mental models of where to look for the pages anyways.
- FAQ and the pages its submenu link to have inconsistent subnavs each so trying to navigate from/around FAQ via subnav you're being taken to places quite randomly;) It's now removed from nav so it's slightly less of an issue, but basically why I mentioned it was that its individual submenus, linking between /faq and /browsers/* seo pages could use a similar tidying up like this one day, to a "stable" shared subnav instead of a handful of oneoffs;)

In an ideal world I would love a breadcrumb+subnav combo;) using the left part for hierarchy, and the right corner for topic-level siblings (think "Firefox › Mobile › Focus" left ~  "Desktop Mobile Feature Compare FAQ" right etc.) to explain the local navigation options a bit more.

I'm basically holding off for the potential nav/masthead redesign before looking into more nav issues myself, so that's just what's been on my mind when it came to nav improvements in general…

But none of that is relevant to this change, so sorry for mixing it up all in one place.

**Review by reemhamz — APPROVED** [review]
🍪🥛 r+!

File: bedrock/firefox/templates/firefox/more.html
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/more.html:1]: Spoke to SEO manager, and she mentioned that since the Enterprise page is linked directly in the products page, that it would be worthwhile in the meantime to also have it share the new Firefox subnav. In the future, if we decide to move the Enterprise page from the main products page, then we'll have it share a subnav with the developer page group and tidy them up. Since these subnav changes are incremental and are currently being worked on, we'll hopefully be seeing more changes come soon

**Pull request merged**
